### PR TITLE
Assorted Japan launch fixes

### PR DESCRIPTION
**Pull request by stephencelis**
A few bug fixes around the Japan launch!

- Selected shipping country fix (while we were rendering the picker with the new localized location name, we were still using the default API "displayable name" once chosen)
- Localized date format fixes (some date formatting was losing localized touches, and the live countdown date format was losing its ",")
- Country code typo fix (was causing no-reward pledges to bottom out at 1¥)

### CODE REVIEW

File: Library/ViewModels/LiveStreamCountdownViewModel.swift [hunk 1]
```diff
@@ -215,9 +215,5 @@ private func flipLiveStreamEventToLive(liveStreamEvent: LiveStreamEvent) -> Live
 
 private func formattedDateString(date: Date) -> String {
 
-  let format = DateFormatter.dateFormat(fromTemplate: "dMMMhmzzz",
-                                        options: 0,
-                                        locale: AppEnvironment.current.locale) ?? "MMM d, h:mm a zzz"
-
-  return Format.date(secondsInUTC: date.timeIntervalSince1970, template: format)
+  return Format.date(secondsInUTC: date.timeIntervalSince1970, template: "dMMMhmzzz")
```
Comment (justinswart) [inline Library/ViewModels/LiveStreamCountdownViewModel.swift]: 🙌

File: Library/ViewModels/RewardPledgeViewModelTests.swift [hunk 2]
```diff
@@ -22,7 +22,7 @@ private let shippingRules = locations
     .template
       |> ShippingRule.lens.location .~ location
       |> ShippingRule.lens.cost .~ Double(idx + 1)
-}
+} ||> ShippingRule.lens.location..Location.lens.localizedName %~ { "Local " + $0 }
```
Comment (justinswart) [inline Library/ViewModels/RewardPledgeViewModelTests.swift]: Doesn't this read slightly better as:
```swift
private let shippingRules = locations
  .enumerated()
  .map { idx, location in
    .template
      |> ShippingRule.lens.location .~ location
      |> ShippingRule.lens.cost .~ Double(idx + 1)
  }
  ||> ShippingRule.lens.location..Location.lens.localizedName %~ { "Local " + $0 }
```
🤔 Dunno, maybe not.

File: KsApi/models/Project.Country.swift [hunk 3]
```diff
@@ -26,7 +26,7 @@ extension Project {
     public static let hk = Country(countryCode: "HK", currencyCode: "HKD", currencySymbol: "$",  maxPledge: 70_000, minPledge: 10, trailingCode: true)
     public static let ie = Country(countryCode: "IE", currencyCode: "EUR", currencySymbol: "€",  maxPledge: 7_000, minPledge: 1, trailingCode: false)
     public static let it = Country(countryCode: "IT", currencyCode: "EUR", currencySymbol: "€",  maxPledge: 7_000, minPledge: 1, trailingCode: false)
-    public static let jp = Country(countryCode: "JA", currencyCode: "JPY", currencySymbol: "¥",  maxPledge: 1_200_000, minPledge: 100, trailingCode: false)
```
Comment (justinswart) [inline KsApi/models/Project.Country.swift]: Hard to keep up with all those ISO standards!

File: Library/Format.swift [hunk 4]
```diff
@@ -273,11 +273,11 @@ private struct DateFormatterConfig {
 
   fileprivate func formatter() -> DateFormatter {
     let formatter = DateFormatter()
+    formatter.locale = self.locale
```
Comment (justinswart) [inline Library/Format.swift]: 👍

File: KsApi/models/Project.Country.swift (hunk 3 above)
Comment (stephencelis) [reply KsApi/models/Project.Country.swift]: Haha, it was a typo. `jp` was the static for a reason, but `ja` is the language code, so confusion!

File: Library/ViewModels/RewardPledgeViewModelTests.swift (hunk 2 above)
Comment (stephencelis) [reply Library/ViewModels/RewardPledgeViewModelTests.swift]: Np! Maybe afterwards since we wanna cut 3.7.1 soon to address these issues.

**Pull request merged**